arcade/arcade/core/errors.py

```diff
@@ -101,3 +110,40 @@ class ToolOutputError(ToolSerializationError):
     """
 
     pass
+
+
+# ------  retry-able errors ------
+
+
+class RetryableToolError(ToolExecutionError):
+    """
+    Raised when a tool error is retryable.
+    """
+
+    _can_retry: bool = True
```

Comment:
To make an error retryable, we simply set `_can_retry` to `True`.

This makes it easier to handle tool execution errors: the developer simply checks for `error.can_retry`, instead of having to differentiate by the type of exception.